dnf/match_counter.py

```diff
@@ -80,9 +74,13 @@ def _key_func(self):
         """
         max_length = self._max_needles()
         def get_key(pkg):
-            return (self._eval_weights(pkg, self[pkg]),
-                    _canonize_string_set(self.matched_needles(pkg), max_length),
-                    -self._eval_distance(pkg, self[pkg]))
+            result = (
+                # use negative value to make sure packages with the highest weight come first
+                - self._eval_weights(pkg, self[pkg]),
+                # then order packages alphabetically
+                pkg,
+            )
+            return result
```

Comment:
I'd prefer to return directly instead of using an intermediate variable... do you prefer this way?

Reply:
it is just my preference. when the return is too long I prefer to make one step more before returning just. I have no problem changing it according to a different coding style

Reply:
I don't think it's a convention, so I'll leave it up to you. It's just my preference to return directly, in this case that even makes the line two characters shorter (in addition to the extra line for the return itself you need) :sunglasses:
```python
return ( ... )
```
 vs
```python
result = ( ... )
return result
```

All other things being equal, I go for less keystrokes :slightly_smiling_face: